src/ehrdata/core/ehrdata.py

```diff
@@ -1,240 +1,129 @@
 from __future__ import annotations
 
+from typing import TYPE_CHECKING
+
+import numpy as np
 import pandas as pd
 from anndata import AnnData
 
 from ehrdata.core.constants import R_LAYER_KEY
 
+if TYPE_CHECKING:
+    from anndata._core.index import Index, Index1D
+
 
-class EHRData:
+class EHRData(AnnData):
     """EHRData object."""
 
+    _t: pd.DataFrame | None
+
     def __init__(
         self,
-        adata=None,
         X=None,
-        r=None,
-        obs=None,
-        var=None,
-        t=None,
-        uns=None,
-        obsm=None,
-        varm=None,
-        layers=None,
-        raw=None,
-        dtype=None,
-        shape=None,
-        filename=None,
-        filemode=None,
-        asview=None,
+        r: np.ndarray | None = None,
         *,
-        obsp=None,
-        varp=None,
-        oidx=None,
-        vidx=None,
-        tidx=None,
+        t: pd.DataFrame | None = None,
+        **kwargs,
     ):
         """EHRData object."""
-        if adata is None:
-            self._adata = AnnData(
-                X=X,
-                obs=obs,
-                var=var,
-                uns=uns,
-                obsm=obsm,
-                varm=varm,
-                layers=layers,
-                raw=raw,
-                dtype=dtype,
-                shape=shape,
-                filename=filename,
-                filemode=filemode,
-                asview=asview,
-                obsp=obsp,
-                varp=varp,
-                oidx=oidx,
-                vidx=vidx,
-            )
-
-        elif isinstance(adata, AnnData):
-            self._adata = adata
-
-        else:
-            raise ValueError("adata must be an AnnData object")
+        super().__init__(X=X, **kwargs)
 
         if r is not None:
+            if (r2 := self.layers.get(R_LAYER_KEY)) is not None and r2 is not r:
+                msg = "`r` is both specified and already present in `adata.layers`."
+                raise ValueError(msg)
```

Comment:
this is a bugfix, I’d say

Reply:
the behavior before would have just overwritten the field I suppose with the r argument?
in any case I did not test the behaviour in this edge case of a properly named layer AND the r handed over too, and this should also not be allowed. good.